asgi_rabbitmq/core.py

```diff
@@ -236,123 +203,39 @@ def publish_properties(self, channel=None):
+            queue=self.get_queue_name(channel),
+        )
 
+    def consume_message(
+        self, future, channel, amqp_channel, method_frame, properties, body
+    ):
         # Don't allow cancel after we got a message
         if future.set_running_or_notify_cancel():
             consumer_tag = method_frame.consumer_tag
-            # Cancel parallel consumers.
-            for tag in consumer_tags:
-                if tag != consumer_tag:
-                    amqp_channel.basic_cancel(consumer_tag=tag)
             amqp_channel.basic_cancel(consumer_tag=consumer_tag)
             amqp_channel.basic_ack(method_frame.delivery_tag)
             # Send the message to the waiting thread.
-            channel = consumer_tags[consumer_tag]
-            consumer_tags.clear()
             if properties.headers and 'asgi_channel' in properties.headers:
                 channel = channel + properties.headers['asgi_channel']
             message = self.deserialize(body)
             future.set_result((channel, message))
 
-    def cancel_consuming(self, future, consumer_tags):
-        for tag in consumer_tags:
-            self.amqp_channel.basic_cancel(consumer_tag=tag)
-        consumer_tags.clear()
-
-    # Non blocking receive.
-
-    def start_non_blocking_receive(self, future, channels):
-        """Sequentially check each channel queue."""
-
-        # In the non-blocking mode get the message from each queue
-        # in series.
-        channel = channels[0]
-        no_message = partial(self.no_message, future, channels[1:])
-        self.amqp_channel.add_callback(no_message, [Basic.GetEmpty])
-        self.amqp_channel.basic_get(
-            partial(self.get_message, future, channel, no_message),
-            queue=self.get_queue_name(channel),
-        )
-
-    def get_message(self, future, channel, no_message, amqp_channel,
-                    method_frame, properties, body):
-        """Message was received in the non-blocking mode."""
-
-        # Cancel no message callback for this queue.
-        amqp_channel.callbacks.remove(
-            amqp_channel.channel_number,
-            Basic.GetEmpty,
-            no_message,
-        )
-        # Send the message to the waiting thread.
-        amqp_channel.basic_ack(method_frame.delivery_tag)
-        if properties.headers and 'asgi_channel' in properties.headers:
-            channel = channel + properties.headers['asgi_channel']
-        message = self.deserialize(body)
-        future.set_result((channel, message))
-
-    def no_message(self, future, channels, method_frame):
-        """
-        The queue is empty.  Tries to read from next channel in
-        non-blocking mode.
-        """
-
-        if channels:
-            self.start_non_blocking_receive(future, channels)
-        else:
-            future.set_result((None, None))
+    def cancel_consuming(self, future, channel):
+        self.amqp_channel.basic_cancel(consumer_tag=channel)
```

Comment:
the argument here should be the value returned from basic_consume not the channel name